sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -398,10 +394,16 @@ public Read withBatchSize(long batchSize) {
 
     @Override
     public void validate(PipelineOptions options) {
+      ConnectionConfiguration connectionConfiguration = getConnectionConfiguration();
       checkState(
```

Comment:
-1. There was a discussion about that in the original PR (https://github.com/apache/beam/pull/1439)  and also in https://issues.apache.org/jira/browse/BEAM-959 (see also related PR). `CheckNotNull` does not generate an explicit enough message to users in that case. And besides the javadoc of `checkNotNull`  specifies "Ensures that an object reference passed as a parameter to the calling method is not null". It is not a parameter in that case.